### PR TITLE
New backend system for Azure Site backend plugin

### PR DESCRIPTION
**Pull request by deepan10**
## Hey, I just made a Pull Request!

Added New backend system for the Azure Sites backend plugin

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/azure-sites-backend/src/index.ts [hunk 1]
```diff
@@ -17,3 +17,4 @@
 export * from './service/router';
 export * from './api';
 export * from './config';
+export { azureSitesPlugin } from './plugin';
```
Comment (awanlin) [inline plugins/azure-sites-backend/src/index.ts:20]:
```suggestion
export { azureSitesPlugin as default } from './plugin';
```

File: plugins/azure-sites-backend/README.md [hunk 2]
```diff
@@ -105,6 +107,29 @@ Here's how to get the backend plugin up and running:
...
-5. Now run `yarn start-backend` from the repo root.
+#### New Backend System
+
+The Azure Sites backend plugin has support for the [new backend system](https://backstage.io/docs/backend-system/), here's how you can set that up:
+
+In your `packages/backend/src/index.ts` make the following changes:
+
+```diff
+  import { createBackend } from '@backstage/backend-defaults';
++ import { azureSitesPlugin } from '@backstage/plugin-azure-sites-backend;
```
Comment (awanlin) [inline plugins/azure-sites-backend/README.md:118]: Remove this line

File: plugins/azure-sites-backend/README.md [hunk 3]
```diff
@@ -105,6 +107,29 @@ Here's how to get the backend plugin up and running:
...
+
+  const backend = createBackend();
+
+  // ... other feature additions
+
++ backend.add(azureSitesPlugin);
```
Comment (awanlin) [inline plugins/azure-sites-backend/README.md:124]:
```suggestion
+ backend.add(import('@backstage/plugin-azure-sites-backend'));
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Cool, I had just assumed this was already one, nice work @deepan10, left some small comments just to follow the existing patterns.

Also, I noticed that in the `README` legacy backend setup section the `catalogApi` is missing from the `createRouter` call. Can you fix that as part of this PR, please? I'm referring to this: https://github.com/backstage/backstage/blob/dde9d0ae753cf1894bd80c48c3001860319255fd/plugins/azure-sites-backend/README.md?plain=1#L56

**Comment by deepan10** [comment]
@awanlin super fast review 🚀 👍  
I will fix the comments that you suggested.

**Review by awanlin — APPROVED** [review]
Awesome, glad to see this added! Many thanks again @deepan10 🚀

File: plugins/azure-sites-backend/README.md [hunk 4]
```diff
@@ -49,14 +51,17 @@ Here's how to get the backend plugin up and running:
    } from '@backstage/plugin-azure-sites-backend';
    import { Router } from 'express';
    import { PluginEnvironment } from '../types';
+   import { CatalogClient } from '@backstage/catalog-client'
 
    export default async function createPlugin(
      env: PluginEnvironment,
    ): Promise<Router> {
      return await createRouter({
+       const catalogApi = new CatalogClient({ discoveryApi: env.discovery })
```
Comment (freben) [inline plugins/azure-sites-backend/README.md:60]:
```suggestion
     const catalogApi = new CatalogClient({ discoveryApi: env.discovery })
     return await createRouter({
```

**Review by freben — APPROVED** [review]
Nice!

**Pull request merged**